pytest_picked/modes.py

```diff
@@ -127,3 +165,36 @@ def parser(self, candidate):
             indicator_index = candidate.find(rename_indicator)
             start_path_index = indicator_index + len(rename_indicator)
         return candidate[start_path_index:]
+
+
+class OnlyChanged(Mode):
```

Comment:
Something went wrong. I tested on a different project where I changed a test without a class and another one under a test and the test class wasn't listed.

Git diff

```
diff --git a/datasets/tests/test_adapters.py b/datasets/tests/test_adapters.py
index 1deea3b..929c665 100644
--- a/datasets/tests/test_adapters.py
+++ b/datasets/tests/test_adapters.py
@@ -77,6 +77,7 @@ def test_save_expense_from_csv():
     assert expense_obj.number == expected_expense["number"]
     assert expense_obj.process_number == expected_expense["process_number"]
     assert expense_obj.value == expected_expense["value"]
+    print(1)


 def test_save_contract_from_csv():
diff --git a/datasets/tests/test_models.py b/datasets/tests/test_models.py
index 3ebb30d..0f09938 100644
--- a/datasets/tests/test_models.py
+++ b/datasets/tests/test_models.py
@@ -18,6 +18,7 @@ from model_bakery import baker

 @pytest.mark.django_db
 class TestCityCouncilAgenda:
+    print("lalala")
     def test_return_none_for_last_collected_item_date_if_nothing_is_found(self):
         # não é salvo no banco - ou seja, banco vazio
         agenda = baker.prepare_recipe("datasets.CityCouncilAgenda")
```

From pytest output:

```
(venv) ➜  maria-quiteria git:(main) ✗ pytest --picked --mode=onlychanged

Changed test files... 1. ['datasets/tests/test_adapters.py::test_save_expense_from_csv']
Changed test folders... 0. []

====== 5 warnings, 1 error in 0.24s =============
ERROR: not found: /home/ana/workspace/maria-quiteria/datasets/tests/test_adapters.py::test_save_expense_from_csv
(no name '/home/ana/workspace/maria-quiteria/datasets/tests/test_adapters.py::test_save_expense_from_csv' in any of [<Module test_adapters.py>])
```

Reply:
git diff
```
diff --git a/tests/foreman/api/test_product.py b/tests/foreman/api/test_product.py
index c7e8b8b..04868d8 100644
--- a/tests/foreman/api/test_product.py
+++ b/tests/foreman/api/test_product.py
@@ -74,6 +74,7 @@ class ProductTestCase(APITestCase):
 
         :CaseImportance: Critical
         """
+        print("")
         label = gen_string('alphanumeric')
         product = entities.Product(label=label, organization=self.org).create()
         self.assertEqual(label, product.label)
diff --git a/tests/foreman/cli/test_ldapauthsource.py b/tests/foreman/cli/test_ldapauthsource.py
index 61dd954..9eb1f3f 100644
--- a/tests/foreman/cli/test_ldapauthsource.py
+++ b/tests/foreman/cli/test_ldapauthsource.py
@@ -485,4 +485,5 @@ class TestRHSSOAuthSource:
                 )
                 ssh_session.prompt()  # match the prompt
                 result = ssh_session.before.decode()
+                print("")
                 assert f"Successfully logged in as '{settings.rhsso.rhsso_user}'." in result
diff --git a/tests/foreman/sys/test_katello_certs_check.py b/tests/foreman/sys/test_katello_certs_check.py
index a63c40e..8b8cbf6 100644
--- a/tests/foreman/sys/test_katello_certs_check.py
+++ b/tests/foreman/sys/test_katello_certs_check.py
@@ -118,7 +118,7 @@ class TestKatelloCertsCheck:
         assert set(options) == expected_result
 
     @pytest.mark.tier1
-    def test_positive_validate_katello_certs_check_output(self, cert_setup, cert_data):
+    def test_positive_validate_katello_certs_check_output_name_change(self, cert_setup, cert_data):
         """Validate that katello-certs-check generates correct output.
 
         :id: 4c9e4c6e-8d8e-4953-87a1-09cb55df3adf
diff --git a/tests/foreman/ui/test_ldap_authentication.py b/tests/foreman/ui/test_ldap_authentication.py
index 0fa4d55..8953741 100644
--- a/tests/foreman/ui/test_ldap_authentication.py
+++ b/tests/foreman/ui/test_ldap_authentication.py
@@ -1852,3 +1852,7 @@ def test_verify_ldap_filters_ipa(session, ipa_add_user, auth_source_ipa, ipa_dat
         with pytest.raises(NavigationTriesExceeded) as error:
             ldapsession.user.search('')
         assert error.typename == "NavigationTriesExceeded"
+        print("")
+
+def test_pass():
+    pass
```
Result 
```
(env) /home/okhatavk/Satellite/robottelo (master) $ pytest --picked --mode=onlychanged --collect-only 

Changed test files... 5. ['tests/foreman/api/test_product.py::ProductTestCase', 'tests/foreman/cli/test_ldapauthsource.py::TestRHSSOAuthSource', 'tests/foreman/sys/test_katello_certs_check.py::TestKatelloCertsCheck', 'tests/foreman/ui/test_ldap_authentication.py::test_verify_ldap_filters_ipa', 'tests/foreman/ui/test_ldap_authentication.py::test_pass']
Changed test folders... 0. []
==================================================================================== test session starts =====================================================================================
platform linux -- Python 3.8.6, pytest-5.4.3, py-1.9.0, pluggy-0.13.1
rootdir: /home/okhatavk/Satellite/robottelo, inifile: pytest.ini
plugins: mock-3.3.1, ibutsu-1.12, services-2.2.1, cov-2.10.1, xdist-1.34.0, forked-1.3.0, picked-0.4.5
collecting ... 2020-12-12 18:09:35 - conftest - DEBUG - Collected 35 test cases
collected 35 items      
```